controllers/infra/capability/infra_capability_controller_unit_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2024 Broadcom. All Rights Reserved.
+// Broadcom Confidential. The term "Broadcom" refers to Broadcom Inc.
+// and/or its subsidiaries.
+
+package capability_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/vm-operator/controllers/infra/capability"
+	pkgcfg "github.com/vmware-tanzu/vm-operator/pkg/config"
+	"github.com/vmware-tanzu/vm-operator/pkg/constants/testlabels"
+	"github.com/vmware-tanzu/vm-operator/test/builder"
+)
+
+func unitTests() {
+	Describe(
+		"Reconcile",
+		Label(
+			testlabels.Controller,
+			testlabels.V1Alpha3,
+		),
+		unitTestsReconcile,
+	)
+}
+
+func unitTestsReconcile() {
+	var (
+		ctx *builder.UnitTestContextForController
+		obj client.ObjectKey
```

Comment:
Nit: I think we can reduce the scope of `obj` and `data` cause they're written to and then read in the same inner blocks.

Reply:
@ridaz Please follow-up with another PR that addresses this nit. Thanks!

Reply:
Hi @akutz @bryanv ,
Please find the PR to incorporate the above review comment:
https://github.com/vmware-tanzu/vm-operator/pull/599/
Thanks